dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationResource.java

```diff
@@ -103,18 +100,53 @@ public final Response getConfigVariables(@Context final HttpServletRequest reque
 				.rejectWhenNoUser(true)
 				.init();
 
-		final String [] keys = StringUtils.splitByCommas(keysQuery);
-		final Map<String,Object> resultMap = new LinkedHashMap<>();
-		if (null != keys) {
+		final Map<String, Object> resultMap = new LinkedHashMap<>();
 
+		if (keysQuery != null && !keysQuery.isEmpty()) {
+			// Old behavior when keys parameter is provided
+			final String[] keys = StringUtils.splitByCommas(keysQuery);
```

Comment:
I think this can be perform with a Stream such as 

Stream.of(key).map(this::removePrefix).filter....